pandas/core/groupby/groupby.py

```diff
@@ -1578,6 +1578,35 @@ def median(self, numeric_only=True):
             numeric_only=numeric_only,
         )
 
+    @final
```

Comment:
is this not just value counts(sort=True, dropna=True)[0]? why are we adding a whole bunch of cython code

Reply:
Well value_counts is about twice as slow as my current implementation(takes same time for 1 column as my implementation does for 2), and all the other methods have corresponding cython code.

Reply:
What is the code that can do `groupby.mode` using `value_counts`?

Reply:
Ah - whoops, I was trying to use DataFrameGroupBy instead of SeriesGroupBy. Ignore the above.

Reply:
> Well value_counts is about twice as slow as my current implementation(takes same time for 1 column as my implementation does for 2), and all the other methods have corresponding cython code.

can you show some timings. I am really hesistant to add this much code for a niche case.